derive/src/lib.rs

```diff
@@ -5,56 +5,58 @@
 
 extern crate proc_macro;
 
-use anyhow::{anyhow, Result};
-use ethabi::{Function, Param, ParamType};
-use ethcontract_common::truffle::Artifact;
-use inflector::Inflector;
-use proc_macro2::{Ident, Literal, Span, TokenStream};
-use quote::quote;
-use std::fs;
+use ethcontract_generate::Builder;
+use proc_macro::TokenStream;
 use syn::ext::IdentExt;
 use syn::parse::{Error as ParseError, Parse, ParseStream, Result as ParseResult};
-use syn::{parse_macro_input, Error as SynError, Ident as SynIdent, LitStr, Token};
+use syn::{parse_macro_input, Error as SynError, Ident, LitStr, Token};
 
 /// Proc macro to generate type-safe bindings to a contract. See
 /// [`ethcontract`](ethcontract) module level documentation for more information.
 #[proc_macro]
-pub fn contract(input: proc_macro::TokenStream) -> proc_macro::TokenStream {
+pub fn contract(input: proc_macro::TokenStream) -> TokenStream {
     let args = parse_macro_input!(input as ContractArgs);
-    expand_contract(args)
-        .unwrap_or_else(|e| SynError::new(Span::call_site(), e.to_string()).to_compile_error())
+    let artifact_span = args.artifact_path.span();
+    args.builder
+        .generate()
```

Comment:
So how exactly would you use the bindings outside of this macro?

Reply:
Docs are incoming! I still have a few things to do in separate PRs:
- Docs for build script
- Build script example
- Refactor `generate/src/contract.rs` into smaller files
- Unit tests for `generate` crate